src/System.Linq/tests/ToListTests.cs

```diff
@@ -135,7 +136,79 @@ public void ToList_WorkWhenCountChangedAsynchronously()
+
+            List<int> emptyIntegersList = new List<int>();
+            List<string> emptyStringsList = new List<string>();
+
+            Assert.Equal(convertedList, sourceIntegers.Select(i => i.ToString()).ToList());
+
+            Assert.Equal(sourceList, sourceIntegers.Where(i => true).ToList());
+            Assert.Equal(emptyIntegersList, sourceIntegers.Where(i => false).ToList());
+
+            Assert.Equal(convertedList, sourceIntegers.Where(i => true).Select(i => i.ToString()).ToList());
+            Assert.Equal(emptyStringsList, sourceIntegers.Where(i => false).Select(i => i.ToString()).ToList());
+
+            Assert.Equal(convertedList, sourceIntegers.Select(i => i.ToString()).Where(s => s != null).ToList());
+            Assert.Equal(emptyStringsList, sourceIntegers.Select(i => i.ToString()).Where(s => s == null).ToList());
+        }
+
+        [Theory]
+        [InlineData(new int[] { }, new string[] { })]
+        [InlineData(new int[] { 1 }, new string[] { "1" })]
+        [InlineData(new int[] { 1, 2, 3 }, new string[] { "1", "2", "3" })]
+        public void ToList_ListWhereSelect(int[] sourceIntegers, string[] convertedStrings)
+        {
+            List<int> sourceList = new List<int>(sourceIntegers);
+            List<string> convertedList = new List<string>(convertedStrings);
+
+            List<int> emptyIntegersList = new List<int>();
+            List<string> emptyStringsList = new List<string>();
+
+            Assert.Equal(convertedList, sourceList.Select(i => i.ToString()).ToList());
+
+            Assert.Equal(sourceList, sourceList.Where(i => true).ToList());
+            Assert.Equal(emptyIntegersList, sourceList.Where(i => false).ToList());
+
+            Assert.Equal(convertedList, sourceList.Where(i => true).Select(i => i.ToString()).ToList());
+            Assert.Equal(emptyStringsList, sourceList.Where(i => false).Select(i => i.ToString()).ToList());
+
+            Assert.Equal(convertedList, sourceList.Select(i => i.ToString()).Where(s => s != null).ToList());
+            Assert.Equal(emptyStringsList, sourceList.Select(i => i.ToString()).Where(s => s == null).ToList());
+        }
+
+        [Theory]
+        [InlineData(new int[] { }, new string[] { })]
+        [InlineData(new int[] { 1 }, new string[] { "1" })]
+        [InlineData(new int[] { 1, 2, 3 }, new string[] { "1", "2", "3" })]
+        public void ToList_IListWhereSelect(int[] sourceIntegers, string[] convertedStrings)
+        {
+            IList<int> sourceList = new ReadOnlyCollection<int>(sourceIntegers);
+            IList<string> convertedList = new ReadOnlyCollection<string>(convertedStrings);
+
+            IList<int> emptyIntegersList = new ReadOnlyCollection<int>(Array.Empty<int>());
+            IList<string> emptyStringsList = new ReadOnlyCollection<string>(Array.Empty<string>());
+
+            Assert.Equal(convertedList, sourceList.Select(i => i.ToString()).ToList());
```

Comment:
Ah, I changed the declarations from `var` to `IList<T>` at the last minute before submitting, and forgot to re-run the tests locally.